### PR TITLE
Turn C48mx500_3DVarAOWCDA back on 

### LINKED ISSUE #2438: Re-enable C48mx500_3DVarAOWCDA test
### What new functionality do you need?

The C48mx500_3DVarAOWCDA had to be disabled as the ocean prep obs job was failing when some categories of obs were not present in dump. Once GDASApp has been updated, the submodule in workflow should be updated and the test turned back.

### What are the requirements for the new functionality?

C48mx500_3DVarAOWCDA completes successfully.

### Acceptance Criteria

- [ ] GDASApp updated to handle missing obs types
- [ ] GDASApp hash updated in workflow
- [ ] C48mx500_3DVarAOWCDA test reenabled on Hera

### Suggest a solution (optional)

_No response_

### LINKED ISSUE #2528: COM_MED_RESTART_PREV is not defined
### What is wrong?

COM_MED_RESTART_PREV in ush/forecast_postdet.sh is not defined https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/forecast_postdet.sh#L698.  This is only defined in scripts/exglobal_stage_ic.sh 

### What should have happened?

COM_MED_RESTART_PREV should be defined

### What machines are impacted?

All or N/A, Hera

### Steps to reproduce

Check out the develop branch, build, link and run the C48mx500_3DVarAOWCDA test, which have now work with updated GSI and GDAS app hashes updated.  I have a branch on hera with this but without some recent commits to develop that show that this test should work. 

### Additional information

This is required to turn back on C48mx500_3DVarAOWCDA

### Do you have a proposed solution?

Perhaps something like the definition in stage ic routine: 
RUN=${rCDUMP} YMD=${gPDY} HH=${gcyc} declare_from_tmpl COM_MED_RESTART_PREV:COM_MED_RESTART_TMP
is needed.

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Description
This PR turns the C48mx500_3DVarAOWCDA test back on.  This required a few bugfixes in GDAS app which have now been merged.    

  Resolves #2438
  Resolves #2528

co-authors: @guillaumevernieres and @aerorahul who provided the bug fixes.  

# Type of change

- Bug fix (fixes something broken, CI test for C48mx500_3DVarAOWCDA)

# Change characteristics
- Is this a breaking change (a change in existing functionality)?  NO
- Does this change require a documentation update? NO

# How has this been tested?
This was tested with the branch of GDAS app hash before it was merged into gdas app develop and a slightly different fix for 2528 on hera.  Additional CI tests should confirm this is acceptable and has no unintended effects on other features. 

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Review by guillaumevernieres — APPROVED** [review]
Thanks @JessicaMeixner-NOAA

**Comment by JessicaMeixner-NOAA** [comment]
For reference, a successful run of the wcda from this morning (g-w copy is not the exact same as here), COMDIR: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_02/C48mx500_3DVarAOWCDA_t03 
g-w: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_02/global-workflow

**Comment by JessicaMeixner-NOAA** [comment]
Running a test from a copy of this PR here: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01

File: ush/forecast_postdet.sh
Comment (JessicaMeixner-NOAA) [inline ush/forecast_postdet.sh]: @aerorahul I know this was not originally the way that you wanted to fix issue #2528 however, the gdas forecast succeeds when I add this.  

Here's the successful log: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/logs/2021032412/gdasfcst.log

I'll start comparing that with a failed log: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/logs/2021032412/gdasfcst.log.0 

to see if we can learn why this line is needed?

Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh]: Okay so from the failed log we get: 

```
+ JGLOBAL_FORECAST[53]: declare_from_tmpl -rx COM_OCEAN_RESTART_PREV:COM_OCEAN_RESTART_TMPL COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
+ bash_utils.sh[35]: [[ NO == \N\O ]]
+ bash_utils.sh[35]: set +x
declare_from_tmpl :: COM_OCEAN_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/ocean/restart
declare_from_tmpl :: COM_MED_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart
```

and then lower down... 

```
+ forecast_postdet.sh[681]: echo 'SUB CMEPS_postdet: Linking output data for CMEPS mediator'
SUB CMEPS_postdet: Linking output data for CMEPS mediator
+ forecast_postdet.sh[683]: [[ .true. == \.\t\r\u\e\. ]]
+ forecast_postdet.sh[686]: local restart_date cmeps_restart_file
+ forecast_postdet.sh[687]: [[ NO == \Y\E\S ]]
+ forecast_postdet.sh[693]: [[ YES == \Y\E\S ]]
+ forecast_postdet.sh[694]: restart_date=2021032409
+ forecast_postdet.sh[698]: cmeps_restart_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[702]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[703]: /bin/cp -p /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t01/gdasfcst.2021032412/fcst.932707/ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[705]: rm -f /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t01/gdasfcst.2021032412/fcst.932707/rpointer.cpl
+ forecast_postdet.sh[706]: touch /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t01/gdasfcst.2021032412/fcst.932707/rpointer.cpl
+ forecast_postdet.sh[707]: echo ufs.cpld.cpl.r.nc
```

and then we fail: 

```
 0:  zeroing coupling accumulated fields at kdt=           10
 0:  zeroing coupling accumulated fields at kdt=           10
 0: -------->(med_phases_restart_read) mediating for: 2021  3 24  9  0  0   0
 0: Abort(1) on node 0 (rank 0 in comm 496): application called MPI_Abort(comm=0x84000003, 1) - process 0
 4: Abort(1) on node 4 (rank 4 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 4
 3: Abort(1) on node 3 (rank 3 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 3
 5: Abort(1) on node 5 (rank 5 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 5
 2: Abort(1) on node 2 (rank 2 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 2
 1: Abort(1) on node 1 (rank 1 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 1
srun: error: h8c48: tasks 0-39: Exited with exit code 1
```


And from the successful log we get: 

```
+ JGLOBAL_FORECAST[53]: declare_from_tmpl -rx COM_OCEAN_RESTART_PREV:COM_OCEAN_RESTART_TMPL COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
+ bash_utils.sh[35]: [[ NO == \N\O ]]
+ bash_utils.sh[35]: set +x
declare_from_tmpl :: COM_OCEAN_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/06//model_data/ocean/restart
declare_from_tmpl :: COM_MED_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/06//model_data/med/restart
```


```
+ forecast_postdet.sh[698]: declare_from_tmpl COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
+ bash_utils.sh[35]: [[ NO == \N\O ]]
+ bash_utils.sh[35]: set +x
declare_from_tmpl :: COM_MED_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart
+ forecast_postdet.sh[699]: cmeps_restart_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[703]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[713]: echo 'WARNING: CMEPS restart file '\''/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc'\'' not found for warm_start='\''.true.'\'', will initialize!'
WARNING: CMEPS restart file '/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc' not found for warm_start='.true.', will initialize!

```


So I guess this is the next question is, which is the correct mediator restart file? And if the "failed" case is the correct restart file, and it does appear to exist, then why isn't the model failing?

Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh]: Okay,, I just re-read the successful run's COM_MED_RESTART_PREV since we do not have 09 cycle's so this is working for the wrong reasons and this is not the correct fix!

Comment (guillaumevernieres) [reply ush/forecast_postdet.sh]: Could  it be that the mediator restart that we have for this test is wrong and that we've been "boot strapping" it without noticing? Why not just erasing the file and let the model initialize the mediator as in your second test above?

Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh]: @guillaumevernieres that might be the case.  I went and looked at a log file from another successful run but before the change for restarts and found: 
```
+ forecast_postdet.sh[980]: local mediator_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[981]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[990]: echo 'WARNING: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc does not exist for warm_start = .true., initializing!'
WARNING: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc does not exist for warm_start = .true., initializing!
+ exglobal_forecast.sh[125]: [[ .true. = .true. ]]
```
That we are not using the mediator restart file... and were looking for the wrong file.  So it's very possible that we needed an updated restart because of updates from a while ago in the model and just never noticed.  This log file is here: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/logs/2021032412/gdasfcst.log

I've looked more at the dates and such and I do think the failed run above without this line was looking for the correct cmeps restart file.  It's very possible that the issue here is that the cplr restart file changed in a previous model version (perhaps a while ago) and we just didn't realize it wasn't correct because we were never looking for the correct file and were starting without it.  

I can try a test run deleting this file to see if things run for this test and report back.  Then, we can decide if we should just remove it or recreate a correct coupler file.

**Comment by JessicaMeixner-NOAA** [comment]
So the likely fastest path to to getting the gdasfcst to run for the first half cycle of C48mx500_3DVarAOWCDA
is to remove the coupler restart file: /scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/gdas.20210324/06/model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc 


(Test confirming this uses ICSDIR_ROOT=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/ICSDIRtest and log file is here: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/logs/2021032412/gdasfcst.log) 

I believe the issue here is that the coupler restart file was made by a previous model version and that the coupler restart is no longer valid for this version.  @aerorahul the scripts already work to just continue forward if this file is not there. 


We theoretically could create a new coupler restart, but I'm not sure that's necessary unless it's testing a capability that should be tested in terms of copying in the coupler restart file. 

There are other failing tests on this PR, but I believe those are also likely failing in develop.   @aerorahul when you get a chance this afternoon, please let me know if there's something more I can do with this PR, or if you (or @guillaumevernieres  or others) think that we should have a coupler restart file, in which case I'll work on getting a new one created.

**Comment by aerorahul** [comment]
Thanks @JessicaMeixner-NOAA 
When `MODE="cycled"`, we could relax the need for the existence of the mediator restart file when initializing the forecast (`gdas` or `gfs`).  
We can write a block, that says this restart is "nice to have" but not required.
If present, it will take it (which it does, but the model fails due to the reasons outlined above; different model version creating the initial mediator restart file, etc).
I think we just put big `TODOs` here to investigate in the future, in the interest of getting this capability of WCDA test running again.

Out of curiosity, did we try running the first half cycle without the mediator restart in COM, and then the subsequent cycles with mediator restarts in COM.  These mediator restarts will have been created by the current model version.  Did this succeed?  If it did, we only need to remove the requirement of the mediator restart at the start of the experiment.

I will look into this and other tests this afternoon.

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul I've only run the first half-cycle so far.  I'm running the rest of the C48mx500_3DVarAOWCDA test now, that's only 1 more cycle but I'll check out the log files to see if the mediator restart is used correctly or not.  We should look into that.  Also, I did not change anything in the code, I only changed what files were present in the ICDIR.

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul The second gdasfcst does grab coupler restarts as we would expect and the full test succeeded.  Log file  /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/logs/2021032418/gdasfcst.log : relevant part: 
```
+ forecast_postdet.sh[698]: cmeps_restart_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/gdas.20210324/12//model_data/med/restart/20210324.150000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[702]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/gdas.20210324/12//model_data/med/restart/20210324.150000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[703]: /bin/cp -p /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/gdas.20210324/12//model_data/med/restart/20210324.150000.ufs.cpld.cpl.r.nc /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t03/gdasfcst.2021032418/fcst.2906429/ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[705]: rm -f /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t03/gdasfcst.2021032418/fcst.2906429/rpointer.cpl
+ forecast_postdet.sh[706]: touch /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t03/gdasfcst.2021032418/fcst.2906429/rpointer.cpl
```

**Comment by aerorahul** [comment]
> @aerorahul The second gdasfcst does grab coupler restarts as we would expect and the full test succeeded. Log file /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/logs/2021032418/gdasfcst.log : relevant part:
> 
> ```
> + forecast_postdet.sh[698]: cmeps_restart_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/gdas.20210324/12//model_data/med/restart/20210324.150000.ufs.cpld.cpl.r.nc
> + forecast_postdet.sh[702]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/gdas.20210324/12//model_data/med/restart/20210324.150000.ufs.cpld.cpl.r.nc ]]
> + forecast_postdet.sh[703]: /bin/cp -p /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test03/COMROOT/c48t03/gdas.20210324/12//model_data/med/restart/20210324.150000.ufs.cpld.cpl.r.nc /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t03/gdasfcst.2021032418/fcst.2906429/ufs.cpld.cpl.r.nc
> + forecast_postdet.sh[705]: rm -f /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t03/gdasfcst.2021032418/fcst.2906429/rpointer.cpl
> + forecast_postdet.sh[706]: touch /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t03/gdasfcst.2021032418/fcst.2906429/rpointer.cpl
> ```

Thanks @JessicaMeixner-NOAA 
This confirms your hypothesis that the mediator restart in the ICs for the test is from an incompatible version of the current model.  We could "skip" this mediator restart when starting the experiment or generate a compatible restart and replace the current one with it.

**Comment by WalterKolczynski-NOAA** [comment]
I've renamed the mediator directories in `C48mx500` on Hera from `med` to `med_no`, so mediator files shouldn't get picked up anymore. Once we confirm tests work, I'll delete them properly and repeat on other machines.

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA there needs to be a med/restart directory, it just has to be empty.  If the directory does not exist I do not think the set up scripts will work. I can do a quick check if needed.

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA there needs to be a med/restart directory, it just has to be empty. If the directory does not exist I do not think the set up scripts will work. I can do a quick check if needed.

This is only at the source they get copied from when setting up an experiment.

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA Have confirmed, we need the restart directory to exist for the experiment to set up: 

```
hfe02:/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test04$ cat README_C48mx500_3DVarAOWCDA 
source ../global-workflow/workflow/gw_setup.sh
pslot=c48t02 RUNTESTS=`pwd` ICSDIR_ROOT=/scratch1/NCEPDEV/global/glopara/data/ICSDIR SLURM_ACCOUNT=marine-cpu ./../global-workflow/workflow/create_experiment.py --yaml ../global-workflow/ci/cases/pr/C48mx500_3DVarAOWCDA.yaml

hfe02:/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test04$ source ../global-workflow/workflow/gw_setup.sh
Running "module reset". Resetting modules to system default. The following $MODULEPATH directories have been removed: None
hfe02:/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test04$ pslot=c48t02 RUNTESTS=`pwd` ICSDIR_ROOT=/scratch1/NCEPDEV/global/glopara/data/ICSDIR SLURM_ACCOUNT=marine-cpu ./../global-workflow/workflow/create_experiment.py --yaml ../global-workflow/ci/cases/pr/C48mx500_3DVarAOWCDA.yaml
2024-04-26 19:05:57,321 - INFO     - root        : BEGIN: __main__.input_args
2024-04-26 19:05:57,321 - DEBUG    - root        : (  )
2024-04-26 19:05:57,323 - INFO     - root        :   END: __main__.input_args
2024-04-26 19:05:57,324 - DEBUG    - root        :  returning: Namespace(yaml=PosixPath('../global-workflow/ci/cases/pr/C48mx500_3DVarAOWCDA.yaml'), overwrite=False)
2024-04-26 19:05:57,338 - INFO     - root        : Call: setup_expt.main()
2024-04-26 19:05:57,338 - DEBUG    - root        : setup_expt.py gfs cycled --pslot c48t02 --app S2S --resdetatmos 48 --resdetocean 5.0 --comroot /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test04/COMROOT --expdir /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test04/EXPDIR --icsdir /scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500 --idate 2021032412 --edate 2021032418 --nens 0 --gfs_cyc 0 --start warm --yaml /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/global-workflow/ci/cases/yamls/soca_gfs_defaults_ci.yaml
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test04/./../global-workflow/workflow/create_experiment.py", line 99, in <module>
    setup_expt.main(setup_expt_args)
  File "/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/global-workflow/workflow/setup_expt.py", line 567, in main
    fill_ROTDIR(host, user_inputs)
  File "/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/global-workflow/workflow/setup_expt.py", line 46, in fill_ROTDIR
    fill_modes[inputs.mode](host, inputs)
  File "/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/global-workflow/workflow/setup_expt.py", line 216, in fill_ROTDIR_cycled
    link_files_from_src_to_dst(src_dir, dst_dir)
  File "/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/global-workflow/workflow/setup_expt.py", line 124, in link_files_from_src_to_dst
    files = os.listdir(src_dir)
            ^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/gdas.20210324/06/model_data/med/restart'
```

**Comment by WalterKolczynski-NOAA** [comment]
Alright, I'll go rename the file then.

**Comment by WalterKolczynski-NOAA** [comment]
Okay, undid the directory name change and renamed the file instead. See if that works.

**Comment by JessicaMeixner-NOAA** [comment]
Thanks @WalterKolczynski-NOAA I can confirm that the experiment will now set-up.  Please let me know if I need to actually run the test or if waiting for CI is sufficient.  

For this PR, it's now expected that the C48mx500_3DVarAOWCDA test will work, however I know that other tests (C96_atmaerosnowDA) were failing and  that's not expected to be changed (yet).  So I don't know what the timeline on testing with CI will be, etc.   I believe those tests also not working in develop either. Please let me know if I can help with that.

**Comment by aerorahul** [comment]
@JessicaMeixner-NOAA 
1. We can let the CI run with the atmaerosnowDA test ON and if it passes, great.
2. If not, we merge this PR, and I will continue investigating and putting an effort into getting that test to run successfully.
3. Or, we can merge now as is, given you are satisfied that this branch with make the AOWCDA test pass.

What would you prefer?

**Comment by JessicaMeixner-NOAA** [comment]
We should run CI tests.

**Comment by WalterKolczynski-NOAA** [comment]
`forecast_postdet` keeps trying to copy a ~mediator~ coupler file that doesn't exist. Does that copy need to be optional?

**Comment by aerorahul** [comment]
> Experiment C96_atmaerosnowDA **FAILED** on Hera in`/scratch1/NCEPDEV/global/CI/2543/RUNTESTS/C96_atmaerosnowDA_b4f68446`

I think this was expected.
The fix for this is in #2534 creating the correct `restart_interval`

**Comment by aerorahul** [comment]
> `forecast_postdet` keeps trying to copy a mediator file that doesn't exist. Does that copy need to be optional?

There is no mediator involved in the `C96_atmaerosnowDA` test.

**Comment by WalterKolczynski-NOAA** [comment]
Sorry, coupler file:
```
+ forecast_postdet.sh[248]: /bin/cp -p /scratch1/NCEPDEV/stmp2/Terry.McGuinness/RUNDIRS/C96_atmaerosnowDA_b4f68446/gdasfcst.2021122012/restart/FV3_RESTART/20211220
.210000.coupler.res /scratch1/NCEPDEV/global/CI/2543/RUNTESTS/COMROOT/C96_atmaerosnowDA_b4f68446/gdas.20211220/12//model_data/atmos/restart/20211220.210000.coupler
.res
/bin/cp: cannot stat '/scratch1/NCEPDEV/stmp2/Terry.McGuinness/RUNDIRS/C96_atmaerosnowDA_b4f68446/gdasfcst.2021122012/restart/FV3_RESTART/20211220.210000.coupler.r
es': No such file or directory
```

**Comment by WalterKolczynski-NOAA** [comment]
All other tests passed, so only `C96_atmaerosnowDA` causing a problem.

**Comment by aerorahul** [comment]
> All other tests passed, so only `C96_atmaerosnowDA` causing a problem.

I opened a PR into this PR's branch in @JessicaMeixner-NOAA's fork https://github.com/JessicaMeixner-NOAA/global-workflow/pull/11
It should fix the failed test.

**Comment by aerorahul** [comment]
> > All other tests passed, so only `C96_atmaerosnowDA` causing a problem.
> 
> I opened a PR into this PR's branch in @JessicaMeixner-NOAA's fork [JessicaMeixner-NOAA#11](https://github.com/JessicaMeixner-NOAA/global-workflow/pull/11) It should fix the failed test.

I am hopeful with this changeset, issue #2524 might also be resolved.  I will continue working on testing the ufs-da tests next.

**Comment by JessicaMeixner-NOAA** [comment]
Thanks for the fixes @aerorahul I have just merged them to this PR!

**Comment by JessicaMeixner-NOAA** [comment]
> All other tests passed, so only `C96_atmaerosnowDA` causing a problem.

@WalterKolczynski-NOAA how do we see that all other tests passed except this one this by the way?

**Comment by WalterKolczynski-NOAA** [comment]
> > All other tests passed, so only `C96_atmaerosnowDA` causing a problem.
> 
> @WalterKolczynski-NOAA how do we see that all other tests passed except this one this by the way?

If you click on `details` next to `continuous-integration/jenkins/pr-head` in the list of checks, it will take you right to the status page for all the steps in the pipeline:

image

You will need to login with your CAC to get into Jenkins (click `ICAM` on the login page). Hovering over the `Matrix...` will show the full name of the case.

**Comment by JessicaMeixner-NOAA** [comment]
> > > All other tests passed, so only `C96_atmaerosnowDA` causing a problem.
> > 
> > 
> > @WalterKolczynski-NOAA how do we see that all other tests passed except this one this by the way?
> 
> If you click on `details` next to `continuous-integration/jenkins/pr-head` in the list of checks, it will take you right to the status page for all the steps in the pipeline:
> 
> image
> 
> You will need to login with your CAC to get into Jenkins (click `ICAM` on the login page). Hovering over the `Matrix...` will show the full name of the case.

@WalterKolczynski-NOAA I was unable to follow your instructions for this.  I will say the last time I did land on the Jenkins page (after finally figuring out this was CAC login), I didn't have permissions to see anything.  Is there a ppt, wiki or readthedocs that goes into this in more detail that might be easier to follow or do I need to find a PR in progess?  (I could not find the continuous integration button you speak of).

**Comment by aerorahul** [comment]
@JessicaMeixner-NOAA Just a follow-up on the Jenkins page; this is a service we use from EPIC and it requires CAC authentication.  EPIC has a whitelist for access, since it allows for changing the behavior of the pipelines, etc.  
We can ask EPIC to allow more users (with CAC) access to the Jenkins controller page.  
Alternately, we can increase the verbosity of the reporting to Github (we are trying to find a balance between too much and too little automated messaging and the content might be subjective based on the user group).

Once the testing is complete (passed), the Jenkins system clears the statuses from the Github PR (at least thats what I have seen)

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul thanks for explaining.  Maybe expanding the verbosity to send a message to have a summary with "x, y, z tests passed" and "a, b, c tests failed"?  The test failure information has really improved and the smaller amount of information has been helpful to not loose things in the PRs.  Thanks for all the work everyone has done to continuously improve this testing process.

**Pull request merged**
